examples/src/zengine_examples/00_Minimal/main.nim

```diff
@@ -1,98 +1,204 @@
 # Minimal example to open up a zengine window
 
-import zengine, sdl2, opengl, glm
+import zengine, opengl, glm, jsbind.emscripten, times, sdl2
 
 
 # Constants
 const
   ScreenWidth = 960
   ScreenHeight = 540
 
-
-# Init zengine
-zengine.init(ScreenWidth, ScreenHeight, "Zengine example: 00_Minimal")
-zengine.gui.init()
-
-
-# State variables
 var
-  # Window control
-  evt = sdl2.defaultEvent
-  running = true
-
-  # Camera control
   camera = Camera(
     position: vec3f(4, 2, 4),
     target: vec3f(0, 1.8, 0),
     up: vec3f(0, 1, 0),
     fovY: 60
   )
-  mouseXRel: int
-  mouseYRel: int
-
-
-# Use a first person camera
-camera.setMode(CameraMode.FirstPerson)
-
-var clock = Timer()
-clock.start()
-
-# Main Game loop
-while running:
-  # Reset
+  mouseWheelMovement = 0
   mouseXRel = 0
   mouseYRel = 0
+  evt: sdl2.Event
+  mainLoopRunning = true
+  model: Model
+  shader: Shader
+
+var gcRequested* = false
+var lastFullCollectTime = 0.0
+const fullCollectThreshold = 128 * 1024 * 1024 # 128 Megabytes	
+template requestGCFullCollect*() =
+  gcRequested = true
+
+proc startApplication() =
+  zengine.init(ScreenWidth, ScreenHeight, "Zengine example: 00_Minimal")
+  start()
+  camera.setMode(CameraMode.Free)
+  shader = loadShader("examples/data/shaders/glsl100/animation/forward.vs", "examples/data/shaders/glsl100/animation/forward.fs")
+  model = loadModel("examples/data/models/mutant/mutant_idle.dae", shader)
+  
+proc runGC() =
```

Comment:
This looks new.  Can you explain it a little more why we're manually taking control of the GC?

Reply:
We're only doing this for emscripten